src/main/java/address/model/BackingAddressBook.java

```diff
@@ -0,0 +1,74 @@
+package address.model;
+
+import address.model.datatypes.ExtractableObservables;
+import address.model.datatypes.Person;
+import address.model.datatypes.ReadablePerson;
+import address.model.datatypes.Tag;
+
+import javafx.collections.FXCollections;
+import javafx.collections.ObservableList;
+
+import java.util.Optional;
+
+/**
+ *
+ */
+class BackingAddressBook implements Model {
+
+    final ObservableList<Person> allPersons;
+    final ObservableList<Tag> allTags;
+
+    {
```

Comment:
This is the instance initializer block that runs right before constructor calls; there cant be anything before the leading brace.